r2ai/interpreter.py

```diff
@@ -799,24 +868,37 @@ def respond(self):
         # https://platform.openai.com/docs/assistants/overview
         if self.openai_client is None:
           self.openai_client = OpenAI()
-        query = []
-        if self.system_message != "":
-          query.append({"role": "system", "content": self.system_message})
-        query.extend(self.messages)
-
-        completion = self.openai_client.chat.completions.create(
-          # TODO: instructions=self.system_message # instead of passing it in the query
-          model=openai_model,
-          max_tokens=maxtokens,
-          temperature=float(self.env["llm.temperature"]),
-          messages=query
-        )
-        response = completion.choices[0].message.content
-        if "content" in self.messages[-1]:
-          last_message = self.messages[-1]["content"]
-        if self.env["chat.reply"] == "true":
-          self.messages.append({"role": "assistant", "content": response})
-        print(response)
+
+        if self.auto_run:
+          if len(self.messages) == 1: 
+            self.messages.insert(0,{"role": "system", "content": SYSTEM_PROMPT_AUTO})
+          response = self.openai_client.chat.completions.create(
```

Comment:
if i understand this correctly. the auto mode is only supported when using the openai model, right? 

maybe we should use something like this for llama? https://github.com/musabgultekin/functionary

Reply:
yeah i'm just merging r2d2 as first pass. I'll check out functionary, but I also had it somewhat working with just prompts at one point last year. It wasn't great unless you use a finetuned for tool calling model.
There's also 'ollama' which is open-source and may be a good thing to support as an endpoint. Has some support for function calls.

Reply:
yeah i know ollama, and i agree that its a quite widely used project and many people and will be nice to have support for it too.

i am also looking forward to publish the r2ai module in pip. so installing the tool will be just: pip install r2ai at some point :) but thtas just another topic.